### PR TITLE
[AC-3752] feat: add function call to carousel adapter interface

### PR DESCRIPTION
**Pull request by bang9**
## Changes
- Added adapter interface for function call data to carousel

ticket: [AC-3752]

## Additional Notes
- 

## Checklist
Before requesting a code review, please check the following:
- [x] **[Required]** CI has passed all checks.
- [x] **[Required]** A self-review has been conducted to ensure there are no minor mistakes.
- [x] **[Required]** Unnecessary comments/debugging code have been removed.
- [x] **[Required]** All requirements specified in the ticket have been accurately implemented.
- [x] Ensure the ticket has been updated with the sprint, status, and story points.

[AC-3752]: https://sendbird.atlassian.net/browse/AC-3752?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -7,10 +7,8 @@ export interface SendbirdChatAICallbacks {
   onWidgetSettingFailure?: (error: Error) => void;
 }
 
-export interface FunctionCallRequestInfo {
-  headers: {
-    'Api-Token': string;
-  };
+export interface FunctionCallRequest {
+  headers: object;
```
Comment (chohongm) [inline src/types.ts:11]: is there any particular reason why type changed to object?

File: src/const.ts [hunk 2]
```diff
@@ -68,18 +68,25 @@ export const DEFAULT_CONSTANT = {
...
+
+        return [];
+      },
+    },
+  },
 } satisfies Partial<Constant>;
 
-type ConfigureSession = (sdk: SendbirdChat | SendbirdGroupChat | SendbirdOpenChat) => SessionHandler;
+function isMealsResponse(response: unknown): response is { meals: { strMeal: string; strMealThumb: string }[] } {
+  return !!response && typeof response === 'object' && 'meals' in response && Array.isArray(response.meals);
```
Comment (chohongm) [inline src/const.ts:86]: !!response <- this is redundant no? typeof response === 'object' should cover it

File: src/const.ts [hunk 3]
```diff
@@ -68,18 +68,25 @@ export const DEFAULT_CONSTANT = {
   messageInputControls: {
     blockWhileBotResponding: 10000,
   },
+  tools: {
+    functionCall: {
+      carouselAdapter({ response }) {
+        if (isMealsResponse(response)) {
+          return response.meals.map((it) => ({ title: it.strMeal, url: '', featured_image: it.strMealThumb }));
```
Comment (chohongm) [inline src/const.ts:75]: can you explain either in ticket or in PR what this meals is?

File: src/const.ts (hunk 2 above)
Comment (bang9) [reply src/const.ts:86]: `null` is an object
```js
// true
typeof null === 'object';
```
so we should use `!!response` to filter it out

File: src/types.ts (hunk 1 above)
Comment (bang9) [reply src/types.ts:11]: The header varies depending on the function call set by the customer, just like other payloads.

File: src/const.ts (hunk 3 above)
Comment (bang9) [reply src/const.ts:75]: I'll share the context via DM

Comment (chohongm) [reply src/const.ts:75]: This logic makes carouselAdapter is only for Noom usecase. Can it be used in other cases in the future? If so, we should make carouselAdapter more generic somehow.

Comment (chohongm) [reply src/const.ts:75]: nvm I think its fine.

**Review by chohongm — APPROVED** [review]
LGTM

File: src/const.ts (hunk 3 above)
Comment (bang9) [reply src/const.ts:75]: Yes, that is an initial value. It can be overridden and will be removed.

**Pull request merged**